recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,238 @@
+        if self.options.multithreading and Version(self.version) >= "1.0.1":
+            print("Multithreading is an expermental feature that can cause segmentation faults. Use it at your own risk!")
+            cmake.definitions["UA_ENABLE_MULTITHREADING"] = self.options.multithreading
+        cmake.definitions["UA_ENABLE_IMMUTABLE_NODES"] = self.options.imutable_nodes
+        cmake.definitions["UA_ENABLE_WEBSOCKET_SERVER"] = self.options.web_socket
+        cmake.definitions["UA_ENABLE_HISTORIZING"] = self.options.historize
+        cmake.definitions["UA_ENABLE_DISCOVERY"] = self.options.discovery
+        cmake.definitions["UA_ENABLE_DISCOVERY_MULTICAST"] = self.options.discovery_multicast
+        cmake.definitions["UA_ENABLE_DISCOVERY_SEMAPHORE"] = self.options.discovery_semaphore
+        cmake.definitions["UA_ENABLE_QUERY"] = self.options.query
+        cmake.definitions["UA_ENABLE_ENCRYPTION"] = self.options.encription
+        cmake.definitions["UA_ENABLE_JSON_ENCODING"] = self.options.json_support
+        if self.options.pub_sub != "None":
+            cmake.definitions["UA_ENABLE_PUBSUB"] = True
+            if self.settings.os == "Linux" and self.options.pub_sub == "Ethernet":
+                cmake.definitions["UA_ENABLE_PUBSUB_ETH_UADP"] = True
+            elif self.settings.os == "Linux" and self.options.pub_sub == "Ethernet_XDP":
+                cmake.definitions["UA_ENABLE_PUBSUB_ETH_UADP_XDP"] = True
+            else:
+                print("PubSub over Ethernet is not supported for your OS!")
+        cmake.definitions["UA_ENABLE_DA"] = self.options.data_access
+        if self.options.compiled_nodeset_descriptions == True:
+            cmake.definitions["UA_ENABLE_NODESET_COMPILER_DESCRIPTIONS"] = self.options.compiled_nodeset_descriptions
+            cmake.definitions["UA_NAMESPACE_ZERO"] = "FULL"
+        else:
+            cmake.definitions["UA_NAMESPACE_ZERO"] = self.options.namescpae_zero
+        cmake.definitions["UA_ENABLE_MICRO_EMB_DEV_PROFILE"] = self.options.embedded_profile
+        cmake.definitions["UA_ENABLE_TYPENAMES"] = self.options.typenames
+        cmake.definitions["UA_ENABLE_STATUSCODE_DESCRIPTIONS"] = self.options.readable_statuscodes
+        cmake.definitions["UA_ENABLE_HARDENING"] = self.options.hardening
+        if self.settings.compiler == "Visual Studio" and self.options.shared == True:
+            cmake.definitions["UA_MSVC_FORCE_STATIC_CRT"] = True
+        if version > "1.0.1":
+            cmake.definitions["UA_COMPILE_AS_CXX"] = self.options.cpp_compatible
+        cmake.configure()
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("LICENSE-CC0", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
```

Comment:
Did you remove the cmake scripts?

Reply:
What do you mean by CMake Scripts? I removed the custom packaging and versioning Cmake modules with a [compatibility patch for v1.0](https://github.com/Hahn-Schickard/conan-center-index/blob/master/recipes/open62541/all/patches/add_conan_center_index_compliance_v_1_0.patch) and [compatibility patch for v1.0.1](https://github.com/Hahn-Schickard/conan-center-index/blob/master/recipes/open62541/all/patches/add_conan_center_index_compliance_v_1_0_1.patch) that open62541 team developed. 

If you are interested here is a list of changes both patches apply: 

* remove git based versioning Cmake module: [tools/cmake/SetGitBasedVersion.cmake](https://github.com/Hahn-Schickard/open62541/blob/master/tools/cmake/SetGitBasedVersion.cmake)
* remove AddSan usage for Clang Debug setups
* remove open62541.pc.in file usage and open62541.pc file generation
* remove open62541_install_tools and open62541_install_nodeset (removes nodeset generation, which requires various shell scripts)
* remove open62541Config.cmake and open62541ConfigVersion.cmake installation
* remove open62541_install_examples
* remove deb [packaging utilities](https://github.com/Hahn-Schickard/open62541/blob/master/tools/packaging/CMakeLists.txt)

Reply:
Oh, I didn't check the patches. Thanks for explaining.

In my rcipes, I remove these files in the conan script instead of patching the sources.
This scales better for newer releases. (For now)